CAIPs/caip-207.md

```diff
@@ -0,0 +1,236 @@
+that conform to CAIP-25 but not to CAIP-207 should be carefully considered and
+accomodated by implementers. 
+
+## Motivation
+
+While some core methods and notifications are foundational to a namespace and
+universally defined out-of-band (meaning all callers and respondents agree to
+them already), others are specific to chains or even to subsets of wallets and
+dapps on a given chain, which requires semantic negotiation and/or network
+routing negotiation before authorization can occur.
+
+## Specification
+
+Two properties are added to the scope objects requested in `optionalScopes` and
+optionally returned in `sessionScopes`.  These are both [ordered] strings of arrays:
+
+1. `rpcEndpoints` is an array of zero or more URLs of RPC endpoints that
+   the caller would prefer the respondent to use, ordered by preference. Each
+   must be a valid URL that addresses an RPC endpoint. The respondent may
+   return it empty, reordered, with less, the same, or even more conformant URLs
+   than received.
+2. `rpcDocuments` is an array of zero or more URLs of machine-readable RPC
+   documents that the caller would prefer the respondent to use, ordered by
+   preference. This set of document collectively defines at least syntactically
+   if not also semantically any methods and/or notifications authorized by the
+   CAIP-25 authorization, in DESCENDING heirarchical authority. (For example,
+   any methods or notifications defined differently by multiple authorites will
+   be interpreted by whichever authority is closer to the 0-index of the array).
+   Each must be a valid URL that addresses a valid openRPC document. The
+   respondent may return it empty, reordered, with less, the same, or even more
+   conformant URLs than received.
+
+### Request
+
+A CAIP-207 request is a valid CAIP-25 except for the two additional properties. 
+
+Example:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_authorization",
+  "params": {
+    "requiredScopes": {
+      "eip155": {
+        "chains": ["eip155:1", "eip155:137"],
+        "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "get_balance", "personal_sign"],
+        "notifications": ["accountsChanged", "chainChanged"]
+      },
+      "eip155:42069": {
+        "methods": ["get_balance", "chainChanged", "42069_sEcReTbAlAnCe"],
+        "rpcDocuments": ["https://openrpc.42069-chain.org/"],
+        "rpcEndpoints": ["https://node1.42069-chain.org/"]
+      },
+      "cosmos": {
+        ...
+      }
+    },
+    "optionalScopes":{
+      "eip155:42161": {
+        "methods": ["eth_sendTransaction", "eth_signTransaction", "get_balance", "personal_sign"],
+        "events": ["accountsChanged", "chainChanged"]
+    },
+    "sessionProperties": {
+      "expiry": "2022-12-24T17:07:31+00:00",
+      "caip154-mandatory": "true"
+    }         
+  }
+}
+```
+
+### Response
+
+The wallet can respond to this method with either a success result or an error message.
```

Comment:
We should probably be defining a test suite for this to make sure implementations are actually conformant. I know this is just a first cut at this but between CAIP-25 and CAIP-207 I see a lot of underspecified states for the protocol occurring between the RPC responder and the RPC requester.

Reply:
couldn't agree more.  I think test vectors might be easier to write into each profile, tho-- i.e.  `/namespaces/eip155/caip211.md#test-vectors`.  Perhaps namespace profiles per-CAIP should have a ##Test-Vectors section instead of an ## Examples section more generally?